internal/entities/properties/properties.go

```diff
@@ -243,3 +243,48 @@ func (p *Properties) Iterate() iter.Seq2[string, *Property] {
 		})
 	}
 }
+
+type PropertyFilter func(key string, prop *Property) bool
+
+func (p *Properties) FilteredCopy(filter PropertyFilter) *Properties {
+	if p == nil {
+		return nil
+	}
+
+	propsMap := xsync.NewMapOf[string, Property]()
+	p.props.Range(func(key string, prop Property) bool {
+		if filter(key, &prop) {
+			propsMap.Store(key, prop)
+		}
+		return true
+	})
+
+	return &Properties{
+		props: propsMap,
+	}
+}
+
+func (p *Properties) Merge(other *Properties) *Properties {
```

Comment:
Probably should have named this MergedCopy or merge in place. It would be easy to change one way or the other.